feat(NotePreview): add NotePreview component for rendering truncated note content

Arnab-Mandal1 opened the pull request:
## Description
Adds a lightweight `NotePreview` component that renders a truncated preview of note content with Tailwind line-clamping styles. The component safely defaults content to an empty string using nullish coalescing and slices content to `props.maxLength` when necessary. JSDoc was added for clarity.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Refactoring
- [ ] Documentation update
- [ ] Test addition/update

## Changes Made
- Added `src/components/notes/NotePreview/NotePreview.tsx` — new component implementing preview truncation and Tailwind `line-clamp` styling.
- Ensured the component uses nullish coalescing for default values and includes JSDoc.
- (If not already present) Added `src/components/notes/NotePreview/NotePreviewProps.ts` for the props interface to follow project conventions.

## Testing Done
- [ ] All tests pass
- [ ] Added new tests
- [x] Manual testing completed (rendered component in the app / Storybook to verify truncation and styling)

## Screenshots (if applicable)
Add screenshots here

## Related Issues
Closes #issue_number

## Checklist
- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Comments added for complex logic (JSDoc present)
- [ ] Documentation updated
- [ ] No console.log or debugging code
- [ ] All tests passing


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a Note Preview component to display note content in a compact, clamped layout for list views.
  * Gracefully handles empty or missing content by defaulting to nothing.
  * Accepts a configurable maximum-length prop for future truncation (prop present but truncation not applied in this release).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Arnab-Mandal1 committed:
feat(NotePreview): add NotePreview component for rendering truncated note content

Arnab-Mandal1 committed:
fix(NotePreviewProps): make content prop optional

dbc2201 reviewed (changes requested):
resolve the converstation

Arnab-Mandal1 committed:
fix(NotePreview): display full content instead of truncated preview

Arnab-Mandal1 committed:
Merge branch 'main' into feature/note-preview

rohit3171999 reviewed (dismissed):
LGTM

Arnab-Mandal1 committed:
Merge branch 'main' into feature/note-preview

Arnab-Mandal1 committed:
Merge branch 'main' into feature/note-preview

Arnab-Mandal1 commented on src/components/notes/NotePreview/NotePreview.tsx:
```diff
@@ -0,0 +1,27 @@
+        ? contents.slice(0, props.maxLength)
+        : contents;
+
+    return (
+        <div className="line-clamp-3 text-base-content">
+            <p>
+                {previewText}
+            </p>
+        </div>
+    );
```
@coderabbitai recheck

Arnab-Mandal1 replied:
check

Arnab-Mandal1 committed:
refactor(NotePreview): remove unused maxLength prop

## Description
Removed the `maxLength` property from the `NotePreview` component's interface and documentation. The component logic does not utilize this property, likely relying on CSS for content truncation, making this prop redundant.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [x] Refactoring
- [ ] Documentation update
- [ ] Test addition/update

## Changes Made
- Removed `maxLength` from the `NotePreviewProps` interface.
- Removed the corresponding `@param props.maxLength` JSDoc entry from `NotePreview.tsx`.

## Testing Done
- [x] All tests pass
- [x] Manual testing completed

## Screenshots (if applicable)
N/A

## Related Issues
N/A

## Checklist
- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] No console.log or debugging code
- [x] All tests passing

Arnab-Mandal1 committed:
Merge remote-tracking branch 'origin/feature/note-preview' into feature/note-preview
